lib/ssm_storage/ssm_storage_db.rb

```diff
@@ -0,0 +1,76 @@
+class SsmStorageDb
+  TABLE_NAME = 'ssm_config_records'.freeze
+  ACTIVE_RECORD_MODEL = 'SsmConfigRecord'.freeze
+  def initialize(file_name)
+    @file_name = file_name
+  end
+
+  def file_exists?
+    active_record_exists = ActiveRecord::Base.connection.table_exists? TABLE_NAME
+    if active_record_exists && (defined? (ACTIVE_RECORD_MODEL.constantize) == 'constant')
+      ACTIVE_RECORD_MODEL.constantize.exists?(:file => @file_name.to_s)
+    else
+      false
+    end
+  end
+
+  def hash
+    match_file = ACTIVE_RECORD_MODEL.constantize.where(:file => @file_name.to_s)
+    hashes = {}
+    match_file.each do |row|
+      hashes[row.accessor_keys] = row.value
+    end
+    hashes = hashes.sort.to_h
+    insert_arrays(reconstruct_hash(hashes)).try(:with_indifferent_access)
+  end
+
+  private
+
+  def reconstruct_hash(hash)
+    final_hash = {}
+    hash.each do |key, value|
+      curr_hash = final_hash
```

Comment:
Why do we need this? can't we just manipulate `filnal_hash`?

Reply:
Since I want to insert a value into our hash given a sequence of keys, I need to be able to get down to deeper levels of the hash, which is why I use `curr_hash` and iteratively go down another level using it while keeping `final_hash`